gadgets/advise_seccomp/README.mdx

```diff
@@ -0,0 +1,485 @@
+---
+title: advise_seccomp
+sidebar_position: 0
+---
+
+import CodeBlock from '@theme/CodeBlock';
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+The seccomp profile advisor gadget records syscalls that are issued in a
+specified container, and then uses this information to generate the corresponding
+seccomp profile.
+
+## Getting started
+
+<Tabs groupId="env">
+    <TabItem value="kubectl-gadget" label="kubectl gadget">
+        ```bash
+        $ kubectl gadget run ghcr.io/inspektor-gadget/gadget/advise_seccomp:%IG_TAG% [flags]
+        ```
+    </TabItem>
+
+    <TabItem value="ig" label="ig">
+        ```bash
+        $ sudo ig run ghcr.io/inspektor-gadget/gadget/advise_seccomp:%IG_TAG% [flags]
+        ```
+    </TabItem>
+</Tabs>
+
+## Flags
+
+No Flags.
+
+## Guide
+
+We need to start the advise_seccomp gadget before running our workload, so it's
+able to capture all the syscalls the container uses.
+
+<Tabs groupId="env">
+<TabItem value="kubectl-gadget" label="kubectl gadget">
+
+```bash
+$ kubectl gadget run advise_seccomp:%IG_TAG% --map-fetch-count=0 --map-fetch-interval=0 --podname default-pod
+```
+
+</TabItem>
+
+<TabItem value="ig" label="ig">
+
+```bash
+$ sudo ig run advise_seccomp:%IG_TAG% --map-fetch-count=0 --map-fetch-interval=0 --containername mycontainer
+```
+
+</TabItem>
+</Tabs>
+
+Then, start our application and interact with it to be sure it generates all the
+syscalls it needs to work.
+
+<Tabs groupId="env">
+<TabItem value="kubectl-gadget" label="kubectl gadget">
+
+import unconfined from '!!raw-loader!./files/seccomp-unconfined.yaml';
+
+<CodeBlock language="yaml">{unconfined}</CodeBlock>
+
+```bash
+$ kubectl apply -f https://raw.githubusercontent.com/inspektor-gadget/inspektor-gadget/refs/heads/%IG_BRANCH%/docs/gadgets/files/seccomp-unconfined.yaml
+```
+
+Use port-forward to access the server from the client machine
+
+```bash
+$ kubectl -n default port-forward default-pod 3000:80 &
+
+$ curl localhost:3000
+<!DOCTYPE html>
+<html>
+<head>
+<title>Welcome to nginx!</title>
+<style>
+html { color-scheme: light dark; }
+body { width: 35em; margin: 0 auto;
+font-family: Tahoma, Verdana, Arial, sans-serif; }
+</style>
+</head>
+<body>
+<h1>Welcome to nginx!</h1>
+<p>If you see this page, the nginx web server is successfully installed and
+working. Further configuration is required.</p>
+
+<p>For online documentation and support please refer to
+<a href="http://nginx.org/">nginx.org</a>.<br/>
+Commercial support is available at
+<a href="http://nginx.com/">nginx.com</a>.</p>
+
+<p><em>Thank you for using nginx.</em></p>
+</body>
+</html>
+```
+
+</TabItem>
+
+<TabItem value="ig" label="ig">
+
+```bash
+$ docker run --name mycontainer --rm -d docker.io/library/nginx
```

Comment:
Shouldn't you run it unconfined?

For this specific workload, this is fine. But if you run an unprivileged container, by default docker will not give CAP_SYS_ADMIN and it block the mount syscall. If the container tries to call `mount`, the advise_seccomp gadget will not capture it because the seccomp filter will block it before it can be captured in `raw_tracepoint/sys_enter`.

Reply:
> If the container tries to call mount, the advise_seccomp gadget will not capture it because the seccomp filter will block it before 

I think it's fine not to record syscalls that are blocked, in any case the application will fail, so it's up to the user to guarantee the application is working fine while recording the syscalls.